support-models/src/main/scala/com/gu/support/promotions/Benefit.scala

```diff
@@ -34,6 +34,29 @@ object IncentiveBenefit {
   implicit val incentiveCodec: Codec[IncentiveBenefit] = deriveCodec
 }
 
+sealed trait IntroductoryPeriodType
+
+case object Issue extends IntroductoryPeriodType
+
+object IntroductoryPeriodType {
+  implicit val decodePeriod: Decoder[IntroductoryPeriodType] = Decoder.decodeString.map(code => fromString(code))
+  implicit val encodePeriod: Encoder[IntroductoryPeriodType] = Encoder.encodeString.contramap[IntroductoryPeriodType](_.toString.toLowerCase)
+
+  private def fromString(s: String) = {
+    s.toLowerCase match {
+      case "issue" => Issue
+    }
+  }
+}
+
+case class IntroductoryPriceBenefit(price: Double, periodLength: Int, periodType: IntroductoryPeriodType) extends Benefit
+
+object IntroductoryPriceBenefit {
+  val jsonName = "introductory_price"
+
+  implicit val incentiveCodec: Codec[IntroductoryPriceBenefit] = deriveCodec
+}
+
 object Benefit {
   implicit val encoder: Encoder[Benefit] = Encoder.instance {
     case d: DiscountBenefit => d.asJson
```

Comment:
Do we need to add `IntroductoryPriceBenefit` here yet?

Reply:
Seems like we should doesn't it? Although it appears to work without, I'll investigate why...